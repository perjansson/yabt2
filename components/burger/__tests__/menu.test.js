import React from 'react'
import { shallow } from 'enzyme'

import Menu from '../menu'

const props = {
    name: 'Bun Meat Bun',
    rank: '4',
    web: 'http://www.bunmeatbun.se',
    showMap: true,
    onToggleBurgerMap: () => {
        console.log('Toggle clicked')
    }
}

describe('Menu', () => {
    it('should render menu', () => {
        const wrapper = shallow(<Menu {...props} />)
        expect(wrapper).toMatchSnapshot()
    })

    it('should call onToggleBurgerMap prop on click', () => {
        const onToggleBurgerMap = jest.fn()
        const wrapper = shallow(
            <Menu {...props} onToggleBurgerMap={onToggleBurgerMap} />
        )

        const toggleLink = wrapper.find(
            'a[data-test-name="toggleBurgerMapLink"]'
        )
        toggleLink.simulate('click')

        expect(onToggleBurgerMap).toHaveBeenCalled()
    })
})
