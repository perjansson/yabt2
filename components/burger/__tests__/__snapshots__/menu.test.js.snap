// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render menu 1`] = `
<React.Fragment>
  <h1>
    Bun Meat Bun
  </h1>
  <Styled(nav)>
    <span>
      rank 
      4
    </span>
    <a
      href="http://www.bunmeatbun.se"
      target="_blank"
    >
      homepage
    </a>
    <a
      data-test-name="toggleBurgerMapLink"
      href="#"
      onClick={[Function]}
      tabIndex="-1"
    >
      map
    </a>
  </Styled(nav)>
</React.Fragment>
`;
